src/main/java/pl/psnc/dei/ui/components/PageNavigationComponent.java

```diff
@@ -101,6 +106,33 @@ private void updateElements() {
         add(button);
     }
 
+    /**
+     * blocking last pages if too much pages, performance restriction
+     *
+     * @return
+     */
+    private Button createLastPageButton(int currentPage, int totalPages) {
+        final Button lastPageButton = createButton(totalPages);
+        boolean enableLastPage = false;
```

Comment:
I would extract the rules for enabling last page to the separate method. And also I would refactor this code by decreasing number of ifs...maybe something more mathematically sophisticated if possible :)